README.md

```diff
@@ -105,63 +111,69 @@ own dependency requirements.
 
 [api_scripts]: src/cc_catalog_airflow/dags/provider_api_scripts
 
-### Setup the Docker way
+### Development setup
 
-The advantage of this method is that it recreates the same environment for your
-testing as is on production.
+You'll need `docker` and `docker-compose` installed on your machine, with
+versions new enough to use version `3` of Docker Compose `.yml` files.
 
-There is a [`Dockerfile`][dockerfile] provided in the [`src/cc_catalog_airflow`][cc_airflow]
-directory. With docker installed, navigate to that directory and run
+To set up environment variables, navigate to the
+[`src/cc_catalog_airflow`][cc_airflow] directory, and run
+```shell
+cp env.template .env
+```
+If needed, fill in API keys or other secrets and variables in `.env`. This is
+not needed if you only want to run the tests. There is a
+[`docker-compose.yml`][dockercompose] provided in the
+[`src/cc_catalog_airflow`][cc_airflow] directory, so from that directory, run
 
 ```shell
-docker build -t cc-catalog-etl  .
+docker-compose up -d
```

Comment:
I'm getting a connection error at this step. It should be possible to bring this up with just the template `.env` file, right?

```
webserver_1  | Traceback (most recent call last):
webserver_1  |   File "/usr/local/airflow/.local/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1248, in _execute_context
webserver_1  |     cursor, statement, parameters, context
webserver_1  |   File "/usr/local/airflow/.local/lib/python3.7/site-packages/sqlalchemy/engine/default.py", line 588, in do_execute
webserver_1  |     cursor.execute(statement, parameters)
webserver_1  | psycopg2.errors.UndefinedTable: relation "connection" does not exist
webserver_1  | LINE 2: FROM connection GROUP BY connection.conn_id
webserver_1  |              ^
webserver_1  | 
webserver_1  | 
webserver_1  | The above exception was the direct cause of the following exception:
webserver_1  | 
webserver_1  | Traceback (most recent call last):
webserver_1  |   File "/usr/local/airflow/.local/bin/airflow", line 37, in <module>
webserver_1  |     args.func(args)
webserver_1  |   File "/usr/local/airflow/.local/lib/python3.7/site-packages/airflow/utils/cli.py", line 75, in wrapper
webserver_1  |     return f(*args, **kwargs)
webserver_1  |   File "/usr/local/airflow/.local/lib/python3.7/site-packages/airflow/bin/cli.py", line 900, in webserver
webserver_1  |     app = cached_app_rbac(None) if settings.RBAC else cached_app(None)
webserver_1  |   File "/usr/local/airflow/.local/lib/python3.7/site-packages/airflow/www/app.py", line 233, in cached_app
webserver_1  |     app = create_app(config, testing)
webserver_1  |   File "/usr/local/airflow/.local/lib/python3.7/site-packages/airflow/www/app.py", line 88, in create_app
webserver_1  |     from airflow.www import views
webserver_1  |   File "/usr/local/airflow/.local/lib/python3.7/site-packages/airflow/www/views.py", line 2443, in <module>
webserver_1  |     class ChartModelView(wwwutils.DataProfilingMixin, AirflowModelView):
webserver_1  |   File "/usr/local/airflow/.local/lib/python3.7/site-packages/airflow/www/views.py", line 2534, in ChartModelView
webserver_1  |     'conn_id': _connection_ids()
webserver_1  |   File "/usr/local/airflow/.local/lib/python3.7/site-packages/airflow/utils/db.py", line 74, in wrapper
webserver_1  |     return func(*args, **kwargs)
webserver_1  |   File "/usr/local/airflow/.local/lib/python3.7/site-packages/airflow/www/views.py", line 2440, in _connection_ids
webserver_1  |     .group_by(Connection.conn_id))]
webserver_1  |   File "/usr/local/airflow/.local/lib/python3.7/site-packages/sqlalchemy/orm/query.py", line 3403, in __iter__
webserver_1  |     return self._execute_and_instances(context)
webserver_1  |   File "/usr/local/airflow/.local/lib/python3.7/site-packages/sqlalchemy/orm/query.py", line 3428, in _execute_and_instances
webserver_1  |     result = conn.execute(querycontext.statement, self._params)
webserver_1  |   File "/usr/local/airflow/.local/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 984, in execute
webserver_1  |     return meth(self, multiparams, params)
webserver_1  |   File "/usr/local/airflow/.local/lib/python3.7/site-packages/sqlalchemy/sql/elements.py", line 293, in _execute_on_connection
webserver_1  |     return connection._execute_clauseelement(self, multiparams, params)
webserver_1  |   File "/usr/local/airflow/.local/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1103, in _execute_clauseelement
webserver_1  |     distilled_params,
webserver_1  |   File "/usr/local/airflow/.local/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1288, in _execute_context
webserver_1  |     e, statement, parameters, cursor, context
webserver_1  |   File "/usr/local/airflow/.local/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1482, in _handle_dbapi_exception
webserver_1  |     sqlalchemy_exception, with_traceback=exc_info[2], from_=e
webserver_1  |   File "/usr/local/airflow/.local/lib/python3.7/site-packages/sqlalchemy/util/compat.py", line 178, in raise_
webserver_1  |     raise exception
webserver_1  |   File "/usr/local/airflow/.local/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1248, in _execute_context
webserver_1  |     cursor, statement, parameters, context
webserver_1  |   File "/usr/local/airflow/.local/lib/python3.7/site-packages/sqlalchemy/engine/default.py", line 588, in do_execute
webserver_1  |     cursor.execute(statement, parameters)
webserver_1  | sqlalchemy.exc.ProgrammingError: (psycopg2.errors.UndefinedTable) relation "connection" does not exist
webserver_1  | LINE 2: FROM connection GROUP BY connection.conn_id
webserver_1  |              ^
webserver_1  | 
webserver_1  | [SQL: SELECT connection.conn_id AS connection_conn_id 
webserver_1  | FROM connection GROUP BY connection.conn_id]
webserver_1  | (Background on this error at: http://sqlalche.me/e/f405)
postgres_1   | 2020-03-17 14:59:20.940 UTC [49] LOG:  unexpected EOF on client connection with an open transaction

```

Reply:
This is due to the fact that as of version 3 `docker-compose.yml` files, the 'condition' (e.g., `healthy`) argument to `depends_on` is no longer supported.  So, the first attempt of the airflow webserver happens before the migrations are complete in the postgres container.

Did it start properly after the first couple of restarts?

Reply:
This is due to the fact that as of version 3 `docker-compose.yml` files, the 'condition' (e.g., `healthy`) argument to `depends_on` is no longer supported. So, the first attempt of the airflow webserver happens before the migrations are complete in the postgres container.

Did it start properly after the first couple of restarts?

Reply:
It seems like it gets stuck in a restart loop, I've let it perform 10+ retries without any luck.

Reply:
Okay, thanks for the heads up.  I'll try to get some sort of waiting for postgres logic working.

Reply:
Great, I'm able to run the tests successfully now.